Not able to start integration from older than current version

## This is a...
<!-- Check ONLY one of the following options with "x" -->
<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ x ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>


## The problem
<!--
Briefly describe the issue you are experiencing (or the feature you want to see implemented on Syndesis).
+ For BUGS, tell us what you were trying to do and what happened instead.
+ For NEW FEATURES, describe the _User Persona_ demanding it and its use case.
-->
It is not possible to publish older than current version of integration. There is no integration build from older version in the OS, it simply deploys the current built version of integration.
The UI looks as if the integration was published from the older version, this is however not true, it still runs the current version.

## Expected behavior
<!-- Describe what the desired behavior would be, enlistin gthe acceptance criteria. -->
When I hit publish on older version - for instance I have "version 1" and "version 2" and I hit "publish" on "version 1", I want the "version 1" to be running

## Screenshot
<!--
For features/bugs tackling with UI functionality, drag and drop a screenshot depicting the desired presentation layer or supporting the UX narrative for the new functionality.
-->
There is a screenshot of the activity log. The "version 1" I created had no logging. I added this only for the "version 2", which means, there is still "version 2" running.

![version_issue](https://user-images.githubusercontent.com/4180208/38976044-49749ace-43b0-11e8-837a-e9c7abd02482.png)

## Tasks involved / Steps to Reproduce
<!--
Enlist all the acceptance criteria for new features or the steps required to reproduce the bug/regression reported.
-->
1. Create integration (and publish) -> this starts as "version 1"
2. Edit the integration (and publish) - e.g. you can simply add logger -> this starts as "version 2"
3. Unpublish the integration using "unpublish" button
4. Publish the "version 1" integration
5. Wait for it to get deployed - in the UI it looks almost as if "version 1" was deployed, but on OS - "version 2" is running, as there was no rebuild

I've managed to reproduce this using steps described by @tplevko in *Steps to Reproduce*. It looks that unpublishing version 2 also removed the config map that's required for version 1 to be published.

From the integration pod:
```
Caused by: java.lang.IllegalArgumentException: PropertiesComponent with name properties must be defined in CamelContext to support property placeholders. Property with key [sql-2.password] not found in properties from text: {{sql-2.password}}
```

--- a/app/server/controller/src/main/java/io/syndesis/server/controller/integration/online/PublishHandler.java
+++ b/app/server/controller/src/main/java/io/syndesis/server/controller/integration/online/PublishHandler.java
@@ -223,9 +223,8 @@ public class PublishHandler extends BaseHandler implements StateChangeHandler {
         }
     }
 
-
-    private Integration integrationOf(IntegrationDeployment integrationDeployment) {
-        return dataManager.fetch(Integration.class, integrationDeployment.getIntegrationId().orElseThrow(() -> new IllegalStateException("IntegrationDeployment doesn't have integration id.")));
+    private static Integration integrationOf(IntegrationDeployment integrationDeployment) {
+        return integrationDeployment.getSpec();
     }
 
     /**

